### PR TITLE
feature: add options to specify desired categories of violations to be reported

### PR DESCRIPTION
**Pull request by gaurav-gogia**

### CODE REVIEW

File: pkg/config/config-reader.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ import (
 	"io/ioutil"
 	"os"
 
-	"github.com/pelletier/go-toml"
+	"github.com/BurntSushi/toml"
```
Comment (kanchwala-yusuf) [inline pkg/config/config-reader.go:24]: Any particular reason we have changed this package. 

This package seems to be unmaintained:
https://github.com/BurntSushi/toml#this-project-is-unmaintained

Comment (gaurav-gogia) [reply pkg/config/config-reader.go:24]: It wasn't really my choice. Maybe @dev-gaur can help here?

Comment (dev-gaur) [reply pkg/config/config-reader.go:24]: tbh I don't remember when I changed to this package. @gaurav-gogia could you try reverting on `github.com/pelletier/go-toml` package and see If everthing works?

Comment (gaurav-gogia) [reply pkg/config/config-reader.go:24]: @dev-gaur sure, will look into it

File: pkg/notifications/notifiers_test.go [hunk 2]
```diff
@@ -55,11 +54,13 @@ func TestNewNotifiers(t *testing.T) {
 			configFile: "notthere",
 			wantErr:    config.ErrNotPresent,
 		},
-		{
-			name:       "invalid toml",
-			configFile: "testdata/invalid.toml",
-			wantErr:    fmt.Errorf("(1, 3): was expecting token =, but got \"am\" instead"),
-		},
+		// this test case keeps failing even if exact same error string is copied from gotErr variable
```
Comment (dev-gaur) [inline pkg/notifications/notifiers_test.go:57]: @gaurav-gogia could you take of these comments?

File: pkg/notifications/notifiers_test.go [hunk 3]
```diff
@@ -44,7 +43,7 @@ func TestNewNotifier(t *testing.T) {
 }
 
 func TestNewNotifiers(t *testing.T) {
-
+	// _, err := NewNotifiers("./testdata/invalid.toml")
```
Comment (dev-gaur) [inline pkg/notifications/notifiers_test.go:46]: @gaurav-gogia could you take of these comments?

File: pkg/policy/opa/engine.go [hunk 4]
```diff
@@ -419,8 +417,8 @@ func (e *Engine) Evaluate(engineInput policy.EngineInput) (policy.EngineOutput,
 	return e.results, nil
 }
 
-// FilterRules will apply the scan and skip rules, and severity level
-func (e *Engine) FilterRules(policyPath string, scanRules, skipRules []string, severity string) {
+// FilterRules will apply the scan and skip rules, severity level and categories
```
Comment (dev-gaur) [inline pkg/policy/opa/engine.go:420]:
```suggestion
// FilterRules will apply the scan and skip rules, categories and severity level
```

This is to maintain the same order as the function signature

File: pkg/runtime/validate.go [hunk 5]
```diff
@@ -103,6 +103,10 @@ func (e *Executor) ValidateInputs() error {
 	}
 	zap.S().Debugf("using policy path %v", e.policyPath)
 
+	if len(e.categories) > 0 && !utils.ValidateCategoryInput(e.categories) {
+		return errSeverityNotSupported
+	}
+	zap.S().Debugf("using categories %v", e.categories)
 	if len(e.severity) > 0 && !utils.ValidateSeverityInput(e.severity) {
```
Comment (dev-gaur) [inline pkg/runtime/validate.go:110]: A blank line before this new if clause would make the code look a bit more near.

File: pkg/policy/opa/engine_test.go [hunk 6]
```diff
@@ -165,13 +166,25 @@ func TestFilterRules(t *testing.T) {
 			assert:      true,
 			regoMapSize: 7,
 		},
```
Comment (dev-gaur) [inline pkg/policy/opa/engine_test.go:168]: @gaurav-gogia regarding the test coverage, you need to cover all the failure test cases ( such as invalid category values, etc.. ) , you need to cover the test cases for the precedence of the flag values over the values passing in config.toml files.. , and you need to cover all the edge cases you can think of.

**Review by dev-gaur — CHANGES_REQUESTED** [review]

File: pkg/notifications/notifiers_test.go (hunk 2 above)
Comment (gaurav-gogia) [reply pkg/notifications/notifiers_test.go:57]: sure ✌️

File: pkg/notifications/notifiers_test.go (hunk 3 above)
Comment (gaurav-gogia) [reply pkg/notifications/notifiers_test.go:46]: sure ✌️

File: pkg/policy/opa/engine_test.go (hunk 6 above)
Comment (gaurav-gogia) [reply pkg/policy/opa/engine_test.go:168]: will look into it

File: pkg/runtime/validate.go (hunk 5 above)
Comment (gaurav-gogia) [reply pkg/runtime/validate.go:110]: sure ✌️

File: pkg/policy/opa/engine.go (hunk 4 above)
Comment (gaurav-gogia) [reply pkg/policy/opa/engine.go:420]: So, should I make change

**from this:**
// FilterRules will apply the scan and skip rules, severity level and categories
`func (e *Engine) FilterRules(policyPath string, scanRules, skipRules, categories []string, severity string)`

**to this:**
// FilterRules will apply the scan and skip rules, categories and severity level
`func (e *Engine) FilterRules(policyPath string, scanRules, skipRules [] string, severity string, categories []string)`

Please advise @dev-gaur

File: pkg/utils/category.go [hunk 7]
```diff
@@ -0,0 +1,51 @@
+package utils
+
+// AcceptedCategories is the list of all policy categories
```
Comment (kanchwala-yusuf) [inline pkg/utils/category.go:3]: @gaurav-gogia , @dev-gaur , can we please update this list of categories with latest list of accepted categories?

**Review by kanchwala-yusuf — CHANGES_REQUESTED** [review]

File: pkg/utils/category.go (hunk 7 above)
Comment (gaurav-gogia) [reply pkg/utils/category.go:3]: @kanchwala-yusuf yes, it is up to date, we have a total of 8 categories and they are all here

**Review by kanchwala-yusuf — APPROVED** [review]
